lib/rspec/rails/matchers/action_mailbox.rb

```diff
@@ -24,7 +24,11 @@ def initialize(message)
 
           def matches?(mailbox)
             @mailbox  = mailbox
-            @receiver = ApplicationMailbox.router.send(:match_to_mailbox, inbound_email)
+            if ApplicationMailbox.router.respond_to?(:mailbox_for)
```

Comment:
We should be conditionally defining the  method based on this.

Reply:
Good idea, I will make the change

Reply:
I switched back because I add and error on `ActionMailbox` and I find it more complicated to duplicated all the block. Maybe you prefer in a method? No problem if you want to push another solution on this branch. :)

On `script/run_build`
```
An error occurred while loading spec_helper.
Failure/Error: if ApplicationMailbox.router.respond_to?(:mailbox_for)

NameError:
  uninitialized constant RSpec::Rails::Matchers::ActionMailbox::ReceiveInboundEmail::ApplicationMailbox
# ./spec/spec_helper.rb:27:in `<top (required)>'
No examples found.
```

Reply:
But we do not want to evaluate every-time `ApplicationMailbox.router.respond_to?(:mailbox_for)` when method is called. I will switch back and fix the error.

Reply:
https://github.com/rspec/rspec-rails/pull/2400/commits/5c4186f58202c177ebc5d783e35e9f15efe2ad7e